sharding/hash_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sharding
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/m3db/m3metrics/metric/id"
+
+	"github.com/spaolacci/murmur3"
+	"github.com/stretchr/testify/require"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func TestHashTypeUnmarshalYAML(t *testing.T) {
+	inputs := []struct {
+		str      string
+		expected HashType
+	}{
+		{str: "murmur32", expected: Murmur32Hash},
+	}
+	for _, input := range inputs {
+		var hashType HashType
+		require.NoError(t, yaml.Unmarshal([]byte(input.str), &hashType))
+		require.Equal(t, input.expected, hashType)
+	}
+}
+
+func TestHashTypeUnmarshalYAMLErrors(t *testing.T) {
+	inputs := []string{
+		"huh",
+		"zero",
+	}
+	for _, input := range inputs {
+		var hashType HashType
+		err := yaml.Unmarshal([]byte(input), &hashType)
+		require.Error(t, err)
+		require.Equal(t, "invalid hash type '"+input+"' valid types are: murmur32", err.Error())
+	}
+}
+
+func TestMurmur32HashShardFn(t *testing.T) {
+	hashType := Murmur32Hash
+	numShards := 1024
+	shardFn, err := hashType.ShardFn()
+	require.NoError(t, err)
+
+	inputs := [][]byte{
+		[]byte("foo"),
+		[]byte("bar"),
+		[]byte("baz"),
+	}
+	for _, input := range inputs {
+		require.Equal(t, murmur3.Sum32(input)%uint32(numShards), shardFn(input, numShards))
```

Comment:
nit: instead of testing that it equals `murmur3.Sum32(input)%uint32(numShards)` perhaps we should test that it equals the actual value returned from that operation with the current package we are using? My concern is that we may bump the version of the package and that could unknowingly cause the output to change which may cause the collector and aggregator to get out of sync.

Reply:
Sounds good will update.